A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js

```diff
@@ -492,23 +492,87 @@ async function loadFuel(mcdu, updateView) {
+            const outerTankFill = Math.min(outerTankCapacity, currentBlockFuelInGallons / 2);
+            if (fuelRAux < outerTankFill && fuelcount === 0) {
+                const fueling = setInterval(function () {
+                    if (fuelRAux >= outerTankFill) {
+                        clearInterval(fueling);
+                    } else {
+                        SimVar.SetSimVarValue(`FUEL TANK RIGHT AUX QUANTITY`, "Gallons", fuelRAux++);
+                    }
+                }, 1800);
+
+            } else if (fuelRAux >= outerTankFill || fuelRAux >= outerTankCapacity) {
+                fuelcount = 1;
+            }
+            if (fuelLAux < outerTankFill && fuelcount === 1) {
+                const fueling = setInterval(function () {
+                    if (fuelLAux >= outerTankFill) {
+                        clearInterval(fueling);
+                    } else {
+                        SimVar.SetSimVarValue(`FUEL TANK LEFT AUX QUANTITY`, "Gallons", fuelLAux++);
+                    }
+                }, 1800);
+            } else {
+                fuelcount = 2;
+                currentBlockFuelInGallons -= fuelLAux + fuelRAux;
+            }
+            const innerTankFill = Math.min(innerTankCapacity, currentBlockFuelInGallons / 2);
+
+            if (fuelRMain < innerTankFill && fuelcount === 2) {
+                const fueling = setInterval(function () {
+                    if (fuelRMain >= innerTankFill) {
+                        clearInterval(fueling);
+                    } else {
+                        SimVar.SetSimVarValue(`FUEL TANK RIGHT MAIN QUANTITY`, "Gallons", fuelRMain++);
+                    }
+                }, 1800);
+            } else {
+                fuelcount = 3;
+            }
+            if (fuelLMain < outerTankFill && fuelcount === 3) {
+                const fueling = setInterval(function () {
+                    if (fuelLMain >= innerTankFill) {
+                        clearInterval(fueling);
+                    } else {
+                        SimVar.SetSimVarValue(`FUEL TANK LEFT MAIN QUANTITY`, "Gallons", fuelLMain++);
+                    }
+                }, 1800);
+            } else {
+                fuelcount = 4;
+                currentBlockFuelInGallons -= fuelLMain + fuelRMain;
+            }
+            const centerTankFill = Math.min(centerTankCapacity, currentBlockFuelInGallons);
+            if (fuelCenter < centerTankFill && fuelcount === 4) {
+                const fueling = setInterval(function () {
+                    if (fuelCenter >= centerTankFill) {
+                        clearInterval(fueling);
+                    } else {
+                        SimVar.SetSimVarValue(`FUEL TANK CENTER QUANTITY`, "Gallons", fuelCenter++);
+                    }
+                }, 1800);
+            } else {
+                fuelcount = 5;
+            }
+            if (fuelcount === 5) {
+                clearInterval(startFuel);
+            }
```

Comment:
**issue**: There's a lot of repetition in this code.
**suggestion**: extract a function which you reuse for the different tanks.

Reply:
Alright will do